src-graphql/client.js

```diff
@@ -109,4 +102,50 @@ export default class Client {
       return response.model.node;
     });
   }
+
+/**
+ * Creates a checkout.
+ *
+ * ```javascript
+ * client.createCheckout({lineItems:[ ... ]}).then(checkout => {
+ *   // do something with the checkout
+ * });
+ * ```
+ *
+ * @method createCheckout
+ * @public
+ * @param {Object} [input] An input object containing zero or more of:
+ *   @param {String} [input.email] An email connected to the checkout
+ *   @param {Array} [input.lineItems] A list of line items in the checkout
+ *   @param {Object} [input.shippingAddress] A shipping address
+ *   @param {String} [input.note] A note for the checkout
+ *   @param {Array} [input.customAttributes] A list of custom attributes
+ * @param {Function} [query] Callback function to specify fields to query on the checkout returned
+ * @return {Promise|GraphModel} A promise resolving with the created checkout.
+ */
+  createCheckout(input = {}, query = checkoutQuery()) {
+    const mutation = this.graphQLClient.mutation((root) => {
+      root.add('checkoutCreate', {args: {input}}, (checkoutCreate) => {
+        checkoutCreate.add('userErrors', (userErrors) => {
+          userErrors.add('message');
+          userErrors.add('field');
+        });
+        query(checkoutCreate, 'checkout');
+      });
+    });
+
+    return this.graphQLClient.send(mutation).then((result) => {
+      // Paginate on line items
+      const promises = [];
+      const lineItems = result.model.checkoutCreate.checkout.lineItems;
+
+      if (lineItems && lineItems.length && lineItems[lineItems.length - 1].hasNextPage.valueOf()) {
```

Comment:
with the new version of the client `valueOf` should no longer be necessary.

However, we should definitely expose a fetchAllPages method with a page size param in `graphql-js-client`